Assignment2/main.cpp

```diff
@@ -0,0 +1,89 @@
+#include <iostream>
+#include <fstream>
+
+
+
+
+int main() {
+
+    std::string menu = "";
+    std::cout << "Enter init or edit" << std::endl;
+    std::cin >> menu;
+    std::cout << "Menu selected" << std::endl;
+    if (menu == "edit") {
+
+        std::string Username = "";
+        std::cout << "Enter your name(first and last)" << std::endl;
+        getline(std::cin, Username);
+
+        std::string Email = "";
+        std::cout << "Enter your Email address" << std::endl;
+        getline(std::cin, Email);
+
+        std::string Pass = "";
+        std::cout << "Enter your password" << std::endl;
+        getline(std::cin, Pass);
+
+        std::string Timezone = "";
+        std::cout << "Enter your timezone" << std::endl;
+        getline(std::cin, Timezone);
+
+        std::string Location = "knownrecipients.txt";
+        std::ofstream textfile;
+        textfile.open(Location);
+        //if (textfile.is_open()){
+        //    std::cout << "file is open" << std::endl;
+        //}
+
+        textfile << Username << std::endl;
+        textfile << Email << std::endl;
+        textfile << Pass << std::endl;
+        textfile << Timezone << std::endl;
+        textfile.close();
+    }else if(menu == "init") {
+
+        std::string Location = "knownrecipients.txt";
+        std::string line = "";
+        std::ifstream myfile(Location);
+        if (myfile.is_open()) {
+            while (getline(myfile, line)) {
+                std::cout << line << std::endl;
+
+            }
+        }
+
+        std::string saved_Username = "";
+        std::string saved_Email = "";
+        std::string saved_Pass = "";
+        std::string saved_Timezone = "";
+
+        std::string Username = "";
+        do {
+            std::cout << "Enter your name(first and last)" << std::endl;
+            getline(std::cin, Username);
+        }while (Username != saved_Username);
+
+        std::string Email = "";
+        do {
+            std::cout << "Enter your Email address" << std::endl;
+            getline(std::cin, Email);
+        }while (Email != saved_Email);
```

Comment:
Since all of your saved_ variables store the same value, I think you can simplify this by writing just one variable storing the empty quotation marks. For example, " std::string empty = "" " and then all of your do loops could be " do{...}while(Email != empty).

Alternatively, there is also another method that can be used to test to make sure that the length of the user input variable is not equal to zero, and with this we could completely remove the empty saved_ string variables.

Good job so far, this is definitely a hard assignment for all of us! I remember you asked in class on Wednesday about using arrays to write the input into the files. If you would like to see that in action, I included a few arrays in my assignment to help with file handling. Feel free to check it out! :)